lib/neuron/connection/http.ex

```diff
@@ -40,33 +40,33 @@ defmodule Neuron.Connection.Http do
 
   defp handle({:ok, response}, json_library, parse_options) do
     case json_library.decode(response.body, parse_options) do
-      {:ok, body} -> build_response(%{response | body: body})
+      {:ok, body} -> build_response_tuple(%{response | body: body})
       {:error, error} -> handle_unparsable(response, error)
       {:error, error, _} -> handle_unparsable(response, error)
     end
   end
 
   defp handle({:error, _} = response, _, _), do: response
 
-  defp build_response(%{status_code: 200} = response) do
+  defp build_response(response) do
```

Comment:
Can we change the order and put `build_response` _after_ `build_response_tuple`? The former is used by the later so that is (imo) easier to read.